hcl2template/functions.go

```diff
@@ -0,0 +1,128 @@
+		"cidrhost":        cidr.HostFunc,
+		"cidrnetmask":     cidr.NetmaskFunc,
+		"cidrsubnet":      cidr.SubnetFunc,
+		"cidrsubnets":     cidr.SubnetsFunc,
+		"coalesce":        stdlib.CoalesceFunc,
+		"coalescelist":    stdlib.CoalesceListFunc,
+		"compact":         stdlib.CompactFunc,
+		"concat":          stdlib.ConcatFunc,
+		"contains":        stdlib.ContainsFunc,
+		"convert":         typeexpr.ConvertFunc,
+		"csvdecode":       stdlib.CSVDecodeFunc,
+		"dirname":         filesystem.DirnameFunc,
+		"distinct":        stdlib.DistinctFunc,
+		"element":         stdlib.ElementFunc,
+		"file":            filesystem.MakeFileFunc(basedir, false),
+		"fileexists":      filesystem.MakeFileExistsFunc(basedir),
+		"fileset":         filesystem.MakeFileSetFunc(basedir),
+		"flatten":         stdlib.FlattenFunc,
+		"floor":           stdlib.FloorFunc,
+		"format":          stdlib.FormatFunc,
+		"formatdate":      stdlib.FormatDateFunc,
+		"formatlist":      stdlib.FormatListFunc,
+		"indent":          stdlib.IndentFunc,
+		"index":           stdlib.IndexFunc,
+		"join":            stdlib.JoinFunc,
+		"jsondecode":      stdlib.JSONDecodeFunc,
+		"jsonencode":      stdlib.JSONEncodeFunc,
+		"keys":            stdlib.KeysFunc,
+		"length":          stdlib.LengthFunc,
+		"log":             stdlib.LogFunc,
+		"lookup":          stdlib.LookupFunc,
+		"lower":           stdlib.LowerFunc,
+		"max":             stdlib.MaxFunc,
+		"md5":             crypto.Md5Func,
+		"merge":           stdlib.MergeFunc,
+		"min":             stdlib.MinFunc,
+		"parseint":        stdlib.ParseIntFunc,
+		"pathexpand":      filesystem.PathExpandFunc,
+		"pow":             stdlib.PowFunc,
+		"range":           stdlib.RangeFunc,
+		"reverse":         stdlib.ReverseFunc,
+		"rsadecrypt":      crypto.RsaDecryptFunc,
+		"setintersection": stdlib.SetIntersectionFunc,
+		"setproduct":      stdlib.SetProductFunc,
+		"setunion":        stdlib.SetUnionFunc,
+		"sha1":            crypto.Sha1Func,
+		"sha256":          crypto.Sha256Func,
+		"sha512":          crypto.Sha512Func,
+		"signum":          stdlib.SignumFunc,
+		"slice":           stdlib.SliceFunc,
+		"sort":            stdlib.SortFunc,
+		"split":           stdlib.SplitFunc,
+		"strrev":          stdlib.ReverseFunc,
+		"substr":          stdlib.SubstrFunc,
+		"timestamp":       pkrfunction.TimestampFunc,
+		"timeadd":         stdlib.TimeAddFunc,
+		"title":           stdlib.TitleFunc,
+		"trim":            stdlib.TrimFunc,
+		"trimprefix":      stdlib.TrimPrefixFunc,
+		"trimspace":       stdlib.TrimSpaceFunc,
+		"trimsuffix":      stdlib.TrimSuffixFunc,
+		"try":             tryfunc.TryFunc,
+		"upper":           stdlib.UpperFunc,
+		"urlencode":       encoding.URLEncodeFunc,
+		"uuidv4":          uuid.V4Func,
+		"uuidv5":          uuid.V5Func,
+		"values":          stdlib.ValuesFunc,
+		"yamldecode":      ctyyaml.YAMLDecodeFunc,
+		"yamlencode":      ctyyaml.YAMLEncodeFunc,
+		"zipmap":          stdlib.ZipmapFunc,
```

Comment:
That's the list of available functions; I didn't import base64sha256, base64sha512,filebase64sha256, filebase64sha512,filemd5,filesha1, filesha256, filesha512, and base64gzip because they are regroupement of functions.